server/src/main/java/keywhiz/service/daos/SecretDAO.java

```diff
@@ -78,18 +79,32 @@
   // to permanently remove secrets
   private static final int MAX_ROWS_REMOVED_PER_TRANSACTION = 1000;
 
-  public SecretDAO(DSLContext dslContext, SecretContentDAOFactory secretContentDAOFactory,
-      SecretSeriesDAOFactory secretSeriesDAOFactory, ContentCryptographer cryptographer) {
+  public SecretDAO(
+      DSLContext dslContext,
+      SecretContentDAOFactory secretContentDAOFactory,
+      SecretSeriesDAOFactory secretSeriesDAOFactory,
+      GroupDAO.GroupDAOFactory groupDAOFactory,
+      ContentCryptographer cryptographer) {
     this.dslContext = dslContext;
     this.secretContentDAOFactory = secretContentDAOFactory;
     this.secretSeriesDAOFactory = secretSeriesDAOFactory;
+    this.groupDAOFactory = groupDAOFactory;
     this.cryptographer = cryptographer;
   }
```

Comment:
The dependency on the Group DAO is new because now Secrets care about owners (Groups).